src/containers/MainPage/MainPage.js

```diff
@@ -33,15 +31,19 @@ class MainPage extends Component {
   onSubmit = () => {
     const { query } = this.state;
     const { history } = this.props;
-    if (query === '' || query === undefined) {
-      this.setState({
-        noResult: true,
-      });
-      return;
+    const errors = this.validate(query);
+    this.setState({ errors });
+    if (!errors.query) {
+      // eslint-disable-next-line react/destructuring-assignment
+      this.props.getResult(query);
```

Comment:
Don't try to ignore eslint errors, it establishes best coding practices.

Reply:
Done